platform/linuxbsd/os_linuxbsd.cpp

```diff
@@ -164,7 +164,12 @@ bool OS_LinuxBSD::_check_internal_feature_support(const String &p_feature) {
 
 String OS_LinuxBSD::get_config_path() const {
 	if (has_environment("XDG_CONFIG_HOME")) {
-		return get_environment("XDG_CONFIG_HOME");
+		if (get_environment("XDG_CONFIG_HOME").is_absolute()) {
+			return get_environment("XDG_CONFIG_HOME");
+		} else {
+			WARN_PRINT_ONCE("`XDG_CONFIG_HOME` is a relative path. Ignoring its value and falling back to `$HOME/.config` or `.` per the XDG Base Directory specification.");
+			return has_environment("HOME") ? get_environment("HOME").plus_file(".config") : ".";
```

Comment:
I'm repeating the logic below here (also in the other functions and classes), but I'm not sure how to avoid it. It might be very obvious, but I can't see how to do it…

Reply:
I missed this, but here's how: #48922.